File: mobile/packages/mobile_scanner/lib/src/enums/android_camera_lens_mode.dart
/// Android-only preference for selecting a specific rear camera lens.
enum AndroidCameraLensMode {
  /// Use the platform's default rear camera selection.
  normal(0),

  /// Prefer the widest available rear camera.
  ultraWide(1);

  const AndroidCameraLensMode(this.rawValue);

  final int rawValue;
}

File: mobile/packages/mobile_scanner/android/src/main/kotlin/dev/steenbakker/mobile_scanner/AndroidCameraLensMode.kt
package dev.steenbakker.mobile_scanner

enum class AndroidCameraLensMode(val rawValue: Int) {
    NORMAL(0),
    ULTRA_WIDE(1);

    companion object {
        fun fromRawValue(rawValue: Int): AndroidCameraLensMode {
            return values().firstOrNull { it.rawValue == rawValue } ?: NORMAL
        }
    }
}
